django/db/models/options.py

```diff
@@ -234,7 +235,7 @@ def _prepare(self, model):
                 if already_created:
                     field = already_created[0]
                 field.primary_key = True
-                self.setup_pk(field)
+                self.setup_pk(field, promoting_parent=True)
```

Comment:
I thought you wanted to differentiate between `auto_created` o2o parent links and the ones that are explicitly defined. It's possible that `parent_link` is enough for your use case.